engine/src/main/java/org/terasology/rendering/nui/internal/NUIManagerInternal.java

```diff
@@ -463,8 +493,16 @@ public void popScreen() {
         return null;
     }
 
-    private <T extends ControlWidget> void initialiseOverlay(T overlay, ResourceUrn screenUri) {
-        InjectionHelper.inject(overlay);
+    private <T extends ControlWidget> void initialiseControlWidget(T overlay, ResourceUrn screenUri) {
+        ContextImpl timedContextForModulesWidgets = new ContextImpl(this.context);
+
+        Module declaringModule = moduleEnvironment.get(screenUri.getModuleName());
+        TypeWidgetLibrary moduleLibrary =
+                new TypeWidgetLibraryImpl(typeWidgetFactoryRegistry, declaringModule, this.context);
+        context.put(TypeWidgetLibrary.class, moduleLibrary);
```

Comment:
What's the rationale behind the creation of a new context and a new type widget library here?

Reply:
Thanks! As long as it works, I don't have a problem. Wanted to know for my own knowledge.